creating_packages/package_approaches.rst

```diff
@@ -164,62 +152,54 @@ This will translate to the cmake variables:
     set(CONAN_LIBS_DEBUG mylibrary_d ${CONAN_LIBS_DEBUG})
     set(CONAN_LIBS_RELEASE mylibrary ${CONAN_LIBS_RELEASE})
 
-And these variables will be correctly applied to each configuration by ``conan_basic_setup()``
-helper.
+And these variables will be correctly applied to each configuration by ``conan_basic_setup()`` helper.
```

Comment:
What about other generators? Maybe it is useful to mention the `cmake_find_package` at least as it is the one we are moving to. I suppose the target created by Conan in the `FindXXX.cmake` uses the proper library depending on the build_type, right?

Reply:
Here I just fixed the outdated information to adequate it to the example, I didn't want to introduce in this PR new information.

I am sure the whole docs would benefit from a review regarding the cmake generators, but we still have to clear the path towards the full usability of the find_package generator.

I will try what you suggest and open a new issue accordingly.

Reply:
I think the only generators currently supported are `cmake` and `visual_studio` (plus their corresponding `multi` variants). we should mention that. 
besides that, what happens to `conan_basic_setup(TARGETS)`, is it supported as well?

Reply:
I agree it deserves an investigation to know if it works or not, but it is not the aim of this PR. @danimtb , if you open the new issue related to this example, please add the concerns by @SSE4 so we have a look into it too.